platform/platform-tests/testSrc/com/intellij/openapi/progress/impl/ProgressIndicatorTest.java

```diff
@@ -856,4 +856,23 @@ private static void checkIndicatorNotThrowInThisOldStaleDisgustingNonCancelableS
     catch (ProcessCanceledException ignored) {
     }
   }
+
+  public void testProgressIndicatorNotOverriddenIn_InRunProcessWithProgressSynchronously_duringWriteAction() {
+    ProgressIndicator progressIndicator = new ProgressIndicatorStub() {
+      @Override
+      public void processFinish() {
+
+      }
+    };
+    ApplicationManager.getApplication().runWriteAction(() -> {
+      ProgressManager.getInstance().executeProcessUnderProgress(() -> {
+        ApplicationManagerEx.getApplicationEx().runProcessWithProgressSynchronously(() -> {
+          ProgressIndicator progressManagerIndicator = ProgressManager.getInstance().getProgressIndicator();
+          assertEquals(progressIndicator, progressManagerIndicator);
```

Comment:
I had a more behavioral test in mind, i.e. you set text on one indicator and check that the other has it as well. Then there would be other ways of fulfilling this requirement, if this one is found to be problematic for any reason.

Reply:
Sure, I had a more behavioral test in my last commit, but thought it may not be easy to understand the test is ultimately ensuring the progress indicator is the one being sent into the executeProcessUnderProgress. Let me go back and rework it a bit.